### DISCUSSION TITLE
Expansion of Latex newcommand "variable."

### DISCUSSION THREAD

**Discussion by Savvylion** [James-Yu/LaTeX-Workshop#3351] [Q&A] [answered]
I am trying to install Latex Workshop into VSCode.  My root file has many included files. To make it easier to manage paths to these files within my latex code, I have defined new commands. Example:  \newcommand{\pathE}{N:/DATA_THUMB/Writing/Publishing/Latex/Books_Typesetting/VAMP_Dissertation/bibliography}.

When I fire up VS Code, Latex Workshop does not appear to be able to expand my newcommand path to the subfolders where I have placed the included files .  Here is the error message: 

[17:49:31] Cannot find .bib file: \pathE/VAMP_Biblio.bib
[17:49:31] Parse fls file.
[17:49:31] Cannot find fls file: n:\DATA_THUMB\Writing\Publishing\Latex\Books_Typesetting\VAMP_Dissertation\mainVAMP.fls
[17:49:31] Could not resolve included file undefined
[17:49:31] Could not resolve included file undefined
[17:49:31] Could not resolve included file undefined
[17:49:31] Could not resolve included file undefined
 
I am attaching a more detailed log of the situation that includes  the output log of Latex Workshop upon VS Code startup, my root file, and screenshots of the directories to show where files are. Errors are highlighted in yellow.

It may be something entirely different. But here is my environment.

VS Code 
Version: 1.68.1 (system setup)
Commit: 30d9c6cd9483b2cc586687151bcbcd635f373630
Date: 2022-06-14T12:48:58.283Z
Electron: 17.4.7
Chromium: 98.0.4758.141
Node.js: 16.13.0
V8: 9.8.177.13-electron.0
OS: Windows_NT x64 10.0.19044
May 2022 (version 1.68)

Windows 10 

Any insight would be greatly appreciated.  Jim Porto 

[LaTeX Workshop Errors_27Jun2022.docx](https://github.com/James-Yu/LaTeX-Workshop/files/8996551/LaTeX.Workshop.Errors_27Jun2022.docx)

**Comment by jlelong** [comment, marked as answer]
The discovery mechanism for included files is detailed at https://github.com/James-Yu/LaTeX-Workshop/wiki/Compile#the-dependencies

When parsing a file, LW never expands macros. This is the compiler's job. Hence, the log message `[17:49:31] Cannot find .bib file: \pathE/VAMP_Biblio.bib`. However, we implemented a second mechanism relying on the presence of the `.fls` file, which is generated by the LaTeX compiler when called with the `-recorder` option (this option is always passed when using `latexmk`). Compile your project once to see that the dependencies are properly discovered and all LW features (in particular completion) work well.

**Reply by Savvylion** [reply]
Thanks. My earlier compiles were breaking and I thought it was a dependency problems, since many of my dependent files were not compiling.   After rebooting VS Code, I got the above error messages and thought it was the continuation of the same problem. I neglected to follow up with a compile, and in some frustration posed the question. I have now compiled and have gotten all my dependencies (TOC, LOT) back, except for biblatex.  I can not get workshop to compile my bibliography using biber, but will continue to work through the issue.
